x/authz/keeper/keeper.go

```diff
@@ -132,19 +147,29 @@ func (k Keeper) DispatchActions(ctx sdk.Context, grantee sdk.AccAddress, msgs []
 }
 
 // SaveGrant method grants the provided authorization to the grantee on the granter's account
-// with the provided expiration time. If there is an existing authorization grant for the
+// with the provided expiration time and insert authorization key into the grants queue. If there is an existing authorization grant for the
 // same `sdk.Msg` type, this grant overwrites that.
 func (k Keeper) SaveGrant(ctx sdk.Context, grantee, granter sdk.AccAddress, authorization authz.Authorization, expiration time.Time) error {
 	store := ctx.KVStore(k.storeKey)
+	skey := grantStoreKey(grantee, granter, authorization.MsgTypeURL())
+
+	grant, found := k.getGrant(ctx, skey)
+	// remove old grant key from the grant queue
+	if found {
```

Comment:
we should only remove from the queue (and insert in line 171) only when grant.Expiration has changed. Otherwise we should only do the normal object save. 
Suggestion, we can define here this variable and reuse it  in line 171
```
shoudUpdateQueue: grant.Expiration != expiration
```